Src/WitsmlExplorer.Frontend/components/ContextMenus/FluidContextMenu.tsx

```diff
@@ -0,0 +1,70 @@
+import OperationContext from "../../contexts/operationContext";
+import { ComponentType } from "../../models/componentType";
+import Fluid from "../../models/fluid";
+import FluidsReport from "../../models/fluidsReport";
+import { createComponentReferences } from "../../models/jobs/componentReferences";
+import { JobType } from "../../services/jobService";
+import { colors } from "../../styles/Colors";
+import ContextMenu from "./ContextMenu";
+import { StyledIcon, menuItemText, onClickDeleteComponents } from "./ContextMenuUtils";
+import { CopyComponentsToServerMenuItem } from "./CopyComponentsToServer";
+import { copyComponents, pasteComponents } from "./CopyUtils";
+import { useClipboardComponentReferencesOfType } from "./UseClipboardComponentReferences";
+
+export interface FluidContextMenuProps {
+  checkedFluids: Fluid[];
+}
+
+const FluidContextMenu = (props: FluidContextMenuProps): React.ReactElement => {
+  const { checkedFluids } = props;
+  const { dispatchOperation } = useContext(OperationContext);
+  const {
+    navigationState: { selectedServer, selectedObject, servers }
+  } = useContext(NavigationContext);
+  const fluidReferences = useClipboardComponentReferencesOfType(ComponentType.Fluid);
+  const selectedFluidsReport = selectedObject as FluidsReport;
+
+  const toDelete = createComponentReferences(
+    checkedFluids.map((fluid) => fluid.uid),
+    selectedFluidsReport,
+    ComponentType.Fluid
+  );
+
+  return (
+    <ContextMenu
+      menuItems={[
+        <MenuItem
+          key={"copy"}
+          onClick={() =>
+            copyComponents(
+              selectedServer,
+              checkedFluids.map((fluid) => fluid.uid),
+              selectedFluidsReport,
+              dispatchOperation,
+              ComponentType.Fluid
+            )
+          }
+          disabled={checkedFluids.length === 0}
+        >
+          <StyledIcon name="copy" color={colors.interactive.primaryResting} />
+          <Typography color={"primary"}>{menuItemText("copy", "fluid", checkedFluids)}</Typography>
+        </MenuItem>,
+        <CopyComponentsToServerMenuItem key={"copyComponentToServer"} componentType={ComponentType.Fluid} componentsToCopy={checkedFluids} />,
+        <MenuItem key={"paste"} onClick={() => pasteComponents(servers, fluidReferences, dispatchOperation, selectedFluidsReport)} disabled={fluidReferences === null}>
+          <StyledIcon name="paste" color={colors.interactive.primaryResting} />
+          <Typography color={"primary"}>{menuItemText("paste", "fluid", fluidReferences?.componentUids)}</Typography>
+        </MenuItem>,
+        <MenuItem key={"delete"} onClick={() => onClickDeleteComponents(dispatchOperation, toDelete, JobType.DeleteComponents)} disabled={checkedFluids.length === 0}>
+          <StyledIcon name="deleteToTrash" color={colors.interactive.primaryResting} />
+          <Typography color={"primary"}>{menuItemText("delete", "fluid", checkedFluids)}</Typography>
+        </MenuItem>
```

Comment:
ShowOnServer menu item can be added here. I see that it is also missing in GeologyIntervalContextMenu, which might have caused the omission.